### PR TITLE
Update ROS launch documentation (frontend documents)

### PR DESCRIPTION
**Pull request by ivanpauno**
Address some comments in https://github.com/ros2/design/pull/163.

### CODE REVIEW

**Comment by ivanpauno** [comment]
Also, added `<executable>` example and renamed after https://github.com/ros2/launch_ros/pull/47

File: articles/151_roslaunch_xml.md [hunk 1]
```diff
@@ -114,6 +114,18 @@ Arguments are limited to the scope of their definition and thus have to be expli
 <arg name="output_path" description="Output path for some processing pipeline"/>
 ```
 
+#### `<executable>` Tag
+
+##### Description
+
+The `<executable>` tag allows for executing any executable in the form of local OS process.
```
Comment (hidmic) [inline articles/151_roslaunch_xml.md]: @ivanpauno consider
```suggestion
The `<executable>` tag allows for executing any executable as a local OS process.
```

File: articles/151_roslaunch_xml.md [hunk 2]
```diff
@@ -227,8 +239,8 @@ All substitutions are enclosed by `$(...)`.
 
 ### Built-in Substitutions
 
-`$(find-package <pkg-name>)`
-: Substituted by the path to package installation directory in the local filesystem.
+`$(find-pkg <pkg-name>)`
+: Substituted by the install prefix path of the package.
```
Comment (hidmic) [inline articles/151_roslaunch_xml.md]: @ivanpauno consider
```suggestion
: Substituted by the install prefix path of the given package.
```
instead.

File: articles/151_roslaunch_xml.md [hunk 3]
```diff
@@ -238,6 +250,11 @@ All substitutions are enclosed by `$(...)`.
   Forward and backwards slashes will be resolved to the local filesystem convention.
   Substitution will fail if the executable cannot be found.
 
+`$(exec-in-package <exec-name> <package-name>)`
+: Substituted by the path to the executable in the local filesystem.
+  Lookups at the `lib` folder of the package.
```
Comment (hidmic) [inline articles/151_roslaunch_xml.md]: @ivanpauno consider
```suggestion
  Executables are looked up in the `lib` directory of the package.
```

**Review by hidmic — APPROVED** [review]
LGTM for a couple tiny nits

**Review by wjwwood — APPROVED** [review]
@hidmic's suggestions make sense, but otherwise lgtm

**Comment by ivanpauno** [comment]
@hidmic I've corrected #163's build problems in https://github.com/ros2/design/pull/247/commits/cb2130d6595f7f0f441985c56bc047021d0db3de (tested locally) and addressed your comments in https://github.com/ros2/design/pull/247/commits/d3e918010030114620b8b8272cae3e4e899046dd.

**Review by hidmic — APPROVED** [review]
LGTM!

**Pull request merged**